Update TUI docs. (#796)
AMDResearch/omniperf: docs/how-to/analyze/tui.rst

.. meta::
   :description: ROCm Compute Profiler analysis: Text-based User Interface

>    :keywords: Omniperf, ROCm, profiler, tool, Instinct, accelerator, GUI, standalone, filter

Reviewer: Remove "Omniperf"?

Author: Users might still be searching "Omniper", do we want omniperf to be part of the search index?

Reviewer: Not a big deal, your choice.
IMO, I kind of, want to remove anything related.

***********************

> Text-based User Interface (TUI) analysis (Beta)

Reviewer 2: It would be better to remove `Beta` from the title and instead add a `notice banner` saying that this feature is in beta version for ROCm 7 release. This banner should be removed in 7.1 or 7.2 release when this feature is stable.

For example: 
<img width="712" alt="{2BFAAB42-CA30-4692-A8D5-46510442F7E7}" src="https://github.com/user-attachments/assets/7a421a0b-9990-42e4-89f0-0b891c51435a" />

.. meta::
   :description: ROCm Compute Profiler analysis: Text-based User Interface
   :keywords: Omniperf, ROCm, profiler, tool, Instinct, accelerator, GUI, standalone, filter

****************************************
Text-based User Interface (TUI) analysis
****************************************

ROCm Compute Profiler's analyze mode now supports a lightweight Text-based User Interface (TUI)

> that provides an interactive terminal experience for enhanced usability. The TUI interface is

Reviewer 3: ```suggestion
that provides an interactive terminal experience for enhanced usability. You can use the TUI interface as
```

> an alternative to the standard CLI if you want to explore analysis results with improved visual

Reviewer 3: ```suggestion
a more visually engaging and interactive alternative to explore analysis results compared to the standard CLI.
```

Reviewer 3: Suggest making it clear if it is the OS CLI or the ROCm Compute Profiler CLI.

> feedback and interactive navigation, but without the additional setup requirements or overhead

Reviewer 3: ```suggestion
It provides enhanced visual feedback and easy navigation without needing the extra setup of a full graphical interface.
```

> of a full graphical interface. This analysis option is implemented as a terminal-based interface

Reviewer 3: ```suggestion
This analysis option is implemented as a terminal-based interface
```

that offers real-time visual feedback, keyboard shortcuts for common actions, and improved
readability with formatted output.

.. note::

>    TUI is currently in beta, so while functional, you may encounter minor issues or limitations.

Reviewer 3: ```suggestion
   TUI is currently in an early access state. While functional, you may encounter minor issues or limitations. Running production workloads is not recommended.
```

Launch the TUI analyzer
----------------------------------

> To launch the ROCm Compute Profiler TUI analyzer, use the ``--tui`` flag with the analysis command.

Reviewer 3: ```suggestion
1. Use the ``--tui`` flag with the analysis command to launch the ROCm Compute Profiler TUI analyzer.
```

For example:

.. code-block:: shell-session

   $ rocprof-compute analyze --tui

> To start the analysis, use the dropdown menu at the top left of the screen to select a single

Reviewer 3: ```suggestion
2. To start the analysis, use the dropdown menu at the top left of the screen to select a single
```

workload from ``rocprof-compute profile`` generated output directories.

.. image:: ../../data/analyze/tui.png
   :align: center
   :alt: ROCm Compute Profiler TUI home screen
   :width: 800

> You should see the center window update with collapsed contents, uncollapse to view tables, charts,

Reviewer 3: ```suggestion
3. You can see the center window update with collapsed contents. Uncollapse to view tables, charts,
```

and graphs visualizing the analysis data.

> Once the analysis results are loaded, you can start interactive analysis with detailed metrics.

Reviewer 3: ```suggestion
4. After the analysis results are loaded, you can start interactive analysis with detailed metrics.
```

4. After the analysis results are loaded, you can start interactive analysis with detailed metrics.
The TUI supports basic keyboard shortcuts, including quit application commands for easy navigation.

TUI analysis structure
----------------------------------

Unlike the :doc:`cli` plain style interfaces, the TUI restructures the analysis workflow into four

> hierarchical chapters to provide a more organized, top-down analysis approach:

Reviewer 3: chapters -> categories

1. Top Stat
2. High Level analysis
3. Detailed block analysis
4. Source Level analysis

> Users are encouraged to follow this top-down hierarchical structure to conduct thorough performance

Reviewer 3: You are recommended to follow this top-down hierarchical structure to conduct a thorough performance analysis, starting with the broad overview and progressively drilling down to specific details.

analysis, starting with the broad overview and progressively drilling down into specific details.

> Current limitations

Reviewer 3: Link this section to the note at the top by adding this text: For the limitation identified, see Current limitation.

----------------------------------
The TUI implementation has several limitations that will be addressed in future releases:

> * PC sampling: Source Level analysis does not have PC sampling enabled by default during the

Reviewer 3: Bold out **PC sampling** and **Filtering capabilities**

  profiling stage. Refer to :doc:`../pc_sampling` for details on how to build and enable PC sampling
  manually.

* Filtering capabilities: Advanced filtering options such as kernel filtering and dispatch

>   filtering are not currently supported. These features will be available in upcoming releases.

Reviewer 3: Advanced filtering options such as kernel filtering and dispatch filtering are currently not supported. These features will be available in upcoming releases.
